d_a_obj_tntrap Matching

WilliamArnett opened the pull request:
Matching for all but the Demo build. The demo build has a regswap

Closes #440

WilliamArnett committed:
99% match

WilliamArnett committed:
Rewrote _execute to increase accuracy. Only issue in .text is a label not loading correctly. I have no idea how to fix this.

WilliamArnett committed:
Rewrote _execute to increase accuracy. Only issue in .text is a label not loading correctly. I have no idea how to fix this.

WilliamArnett committed:
Equivalent. Weak func ordering.

WilliamArnett committed:
Equivalent. Weak func ordering.

WilliamArnett committed:
7% text

WilliamArnett committed:
28% text

WilliamArnett committed:
41% text

WilliamArnett committed:
65% text

WilliamArnett committed:
merge in changes to main

WilliamArnett committed:
82% text

WilliamArnett committed:
Equivalent. A lot of work is needed to name variables and clean up functions. Weak function ordering is wrong.

WilliamArnett committed:
Cleaned some functions, named several variables, and began documenting dialog tree.

WilliamArnett committed:
Naming a couple more variables

WilliamArnett committed:
21% text

WilliamArnett committed:
change out getMySubstanceP for getMyIntegerP inline

WilliamArnett committed:
100% text

WilliamArnett committed:
Refactor to use JUT_ASSERT macro

WilliamArnett committed:
Merge pull request #2 from WilliamArnett/d_a_npc_kk1

D a npc kk1

WilliamArnett committed:
Merge branch 'main' into d_a_npc_km1

WilliamArnett committed:
Merge pull request #3 from WilliamArnett/d_a_npc_km1

D a npc km1

WilliamArnett committed:
Merge pull request #4 from zeldaret/main

D_a_obj_pfall matching (#769)

WilliamArnett committed:
25% text

WilliamArnett committed:
58% text. Added inline func to d_a_ship.h

WilliamArnett committed:
Merge branch 'main' of https://github.com/zeldaret/tww into d_a_obj_tntrap

WilliamArnett committed:
d_a_obj_tntrap Equivalent

WilliamArnett committed:
Matching for GZLJ01, GZLE01, GZLP01

WilliamArnett committed:
Merge branch 'main' of https://github.com/zeldaret/tww into d_a_obj_tntrap

Skelp committed:
#181: d_a_tag_attention matching; TODO: implement chk_inside and dComIfGp_att_Look2RequestF

Skelp committed:
Merge branch 'zeldaret:main' into main

ammaraskar commented on src/d/actor/d_a_obj_tntrap.cpp:
```diff
@@ -4,172 +4,643 @@
+#include "m_Do/m_Do_mtx.h"
+
+
+const char l_arcname[] = "TnTrap";
```
These variables should be inside an anonymous namespace based on the original symbol names.

ammaraskar commented on src/d/actor/d_a_obj_tntrap.cpp:
```diff
@@ -4,172 +4,643 @@
+        }
+        break;
+    default:   
+        JUT_ASSERT(0x17C,0);
```
```suggestion
        JUT_ASSERT(380, FALSE);
```

Macros get expanded in the assertions, this was probably just `FALSE`. First argument to the `JUT_ASSERT` macro is a line number so keeping it in decimal might make more sense.

ammaraskar commented on src/d/actor/d_a_obj_tntrap.cpp:
```diff
@@ -4,172 +4,643 @@
-    /* Nonmatching */
+bool daObjTnTrap_c::create_heap() {
+    bool o_retval = true;
+    cBgD_t* pcVar1 = (cBgD_t*)dComIfG_getObjectRes(l_arcname,0x3);
```
```suggestion
    cBgD_t* pcVar1 = (cBgD_t*)dComIfG_getObjectRes(l_arcname,TNTRAP_DZB_TN_WALL01);
```

ammaraskar commented on src/d/actor/d_a_obj_tntrap.cpp:
```diff
@@ -4,172 +4,643 @@
+    if(field_0xDC0 >= 5 || field_0xDC0 < 1){
+        return;
+    }
+    fopAcM_seStartCurrent(this,0x6239,0);
```
```suggestion
    fopAcM_seStartCurrent(this, JA_SE_OBJ_TN_TRAP, 0);
```

ammaraskar commented on src/d/actor/d_a_obj_tntrap.cpp:
```diff
@@ -4,172 +4,643 @@
+
 }
 
+u32 fillerbytes[12] = {
```
Is to make the data section match up? Check out this part of the tutorial, one of the weak data/bss includes might be able to fix it instead: https://github.com/zeldaret/tww/blob/main/docs/decompiling.md#missing-weak-data

ammaraskar commented on src/d/actor/d_a_obj_tntrap.cpp:
```diff
@@ -4,172 +4,643 @@
+                int var_29 = 4;
+                if(field_0xDD0 == 0){
+                    var_29 = 2;
+                    dComIfGs_onEventBit(0x3B40);
```
Same comment about the event constant here.

ammaraskar commented on src/d/actor/d_a_obj_tntrap.cpp:
```diff
@@ -4,172 +4,643 @@
+    if (field_0xDC4 == 1) {
+    dComIfG_resDelete(&field_0x290,l_arcname);
+        if (heap != NULL){
+            if(field_0xD58 != NULL) {
```
You're missing an inline here, this is:

```cpp
if (heap != NULL && field_0xD58 != NULL) {
  if (field_0xD58->ChkUsed()) {
      dComIfG_Bgsp()->Release(field_0xD58);
  }
  field_0xD58 = NULL;
}

ammaraskar commented on src/d/actor/d_a_obj_tntrap.cpp:
```diff
@@ -4,172 +4,643 @@
+            case 1:
+                break;
+            case 0:
+                mDoAud_seStart(0x806);
```
```suggestion
                mDoAud_seStart(JA_SE_READ_RIDDLE_1);
```

ammaraskar commented on src/d/actor/d_a_obj_tntrap.cpp:
```diff
@@ -4,172 +4,643 @@
 bool daObjTnTrap_c::_execute() {
-    /* Nonmatching */
+
+    if(field_0xD58 != NULL){
```
can consolidate this to:

```cpp
if (field_0xD58 && field_0xD58->ChkUsed()) {
```

ammaraskar commented on src/d/actor/d_a_obj_tntrap.cpp:
```diff
@@ -4,172 +4,643 @@
+    field_0xDD4 = param_get_mapType();
+    switch(field_0xDD4){
+    case 0:
+        if(dComIfGs_isEventBit(0x3A04) == 1){
```
I know there's places where event values are hardcoded like this, but it might make more sense to make an enum under `class dSv_evtBit_c` like `EVT_UNK_3A04` just so it's consolidated in one place when we go to figure out all the event ids.

WilliamArnett replied:
Ah, that's a good idea. I will do so 👍

ammaraskar commented on src/d/actor/d_a_obj_tntrap.cpp:
```diff
@@ -4,172 +4,643 @@
+        /* Flags             */ 0,
+        /* SrcObjAt  Type    */ 0x0800,
+        /* SrcObjAt  Atp     */ 1,
+        /* SrcObjAt  SPrm    */ 5,
```
```suggestion
        /* SrcObjAt  SPrm    */ cCcD_AtSPrm_Set_e | cCcD_AtSPrm_VsPlayer_e,
```

ammaraskar commented on src/d/actor/d_a_obj_tntrap.cpp:
```diff
@@ -4,172 +4,643 @@
+static const dCcD_SrcTri l_tri_src = {
+    {
+        /* Flags             */ 0,
+        /* SrcObjAt  Type    */ 0x0800,
```
```suggestion
        /* SrcObjAt  Type    */ AT_TYPE_UNK800,
```

ammaraskar commented on src/d/actor/d_a_obj_tntrap.cpp:
```diff
@@ -4,172 +4,643 @@
+        /* SrcObjCo  SPrm    */ 0,
+        /* SrcGObjAt Se      */ 0,
+        /* SrcGObjAt HitMark */ 0x00,
+        /* SrcGObjAt Spl     */ 0xB,
```
```suggestion
        /* SrcGObjAt Spl     */ dCcG_At_Spl_UNKB,
```

Skelp committed:
#181: continue working on d_a_tag_attention, matching almost complete

Skelp committed:
Merge branch 'main' of https://github.com/Skelp/tww

Skelp committed:
#181: revert naming class member (not certain enough)

Skelp committed:
Merge remote-tracking branch 'upstream/main'

Skelp committed:
#181: preliminary changes to resolve #784 comments regarding usage of debug maps

WilliamArnett committed:
Implement suggestions from ammaraskar

WilliamArnett committed:
Merge branch 'main' of https://github.com/zeldaret/tww into d_a_obj_tntrap

WilliamArnett committed:
Revert config.yml

Unused data export for d_a_obj_tntrap

Skelp committed:
#181: commit suggestion by @WilliamArnett, fixing regalloc error

Co-authored-by: WilliamArnett <149556961+WilliamArnett@users.noreply.github.com>

Skelp committed:
#181: fix missing curly bracket

Skelp committed:
Merge remote-tracking branch 'upstream/main'

Skelp committed:
#181: resolve #784 comments with the kind help of @WilliamArnett;
 move `daTagAttention::Act_c::chk_inside` to header file; split `plyrToObjVec` definition to match assembly;
 remove temp variable, use values directly and swap operands;
 implement `Look2RequestF` in `include/d/d_attention.h`;
 remove non-matching use of temp variable `iVar1` and replace `current.pos` with uninitialized variable in `daTagAttention::Act_c::_execute`

WilliamArnett committed:
Implement demo HIO class

WilliamArnett committed:
Merge branch 'main' of https://github.com/zeldaret/tww into d_a_obj_tntrap

WilliamArnett committed:
test particle_set refactor

WilliamArnett committed:
More demo matching. Name fields and implement action enum

WilliamArnett committed:
JUT_ASSERT line numbers in decimal

WilliamArnett committed:
Remove demo match from configure.py

WilliamArnett committed:
Merge branch 'main' of https://github.com/zeldaret/tww

WilliamArnett committed:
Merge branch 'main' of https://github.com/WilliamArnett/tww

WilliamArnett commented:
The tu in 'tutrap' is an abbreviation of タートナック

WilliamArnett committed:
Merge remote-tracking branch 'upstream/main'

WilliamArnett committed:
Update fork to match upstream

WilliamArnett committed:
Merge branch 'main' into d_a_obj_tntrap

WilliamArnett committed:
revert d_a_npc_kk1.h changes

WilliamArnett committed:
Delete debug/m2cdecomp

WilliamArnett committed:
revert d_a_npc_kk1.cpp and d_a_npc_km1.cpp

WilliamArnett committed:
Merge branch 'd_a_obj_tntrap' of https://github.com/WilliamArnett/tww into d_a_obj_tntrap

WilliamArnett committed:
Update d_a_tag_attention.cpp

WilliamArnett committed:
reimplement TNTRAP_ACTION_IDX enum

WilliamArnett committed:
Merge branch 'd_a_obj_tntrap' of https://github.com/WilliamArnett/tww into d_a_obj_tntrap
